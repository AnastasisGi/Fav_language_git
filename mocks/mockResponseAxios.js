const response = [
	{ id: 1, language: 'Ruby' },
	{ id: 2, language: 'Javascript' },
	{ id: 3, language: 'Ruby' },
];

export default response;
