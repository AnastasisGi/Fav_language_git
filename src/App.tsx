import React from 'react';
import logo from './logo.svg';
import './App.css';
import Form from './Components/Form';
import Results from './Components/Results';
import { useState } from 'react';
import Header from './Components/Header';

interface Iinfo {
  username: string;
  language: string;
}



function App() {
  const [info, setInfo] = useState<Iinfo>({ username: '', language: '' });
  const [displayResultBlock, setDisplayResultBlock] = useState<boolean>(false)
  return (
    <div className="App">
      <Header />

      <Form setInfo={setInfo}
      setDisplayResultBlock={setDisplayResultBlock}
      />
      <br></br>
      <Results info={info} 
      displayResultBlock={displayResultBlock} />
    </div>
  );
}

export default App;
