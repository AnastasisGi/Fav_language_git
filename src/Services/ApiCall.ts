import axios from 'axios';

const ApiCall = async (username: string) => {
	try {
		console.log('apicall started');
		const  data  = await axios.get(
			`https://api.github.com/users/${username}/repos?per_page=200`
		).then(response=>response.data)
                // console.log('DATTTA',data);
		return gatherLanguages(data);
	} catch (error) {
		console.log('error', error);
	}
};

function gatherLanguages(data: any) {
	let newArray: string[] = [];
	data.forEach((prj: any) => {
		// prj.language === null ? return : newArray.push[prj.language]
		if (prj.language === null) return;

		// eslint-disable-next-line @typescript-eslint/no-unused-expressions
		newArray.push(prj.language);
	});

	return getMostFrequent(newArray);
}

function getMostFrequent(newArray: string[]) {
	const hashmap = newArray.reduce(
		(acc: { [key: string]: number }, val: string) => {
			acc[val] = (acc[val] || 0) + 1;

			return acc;
		},
		{}
	);
	return Object.keys(hashmap).reduce((a, b) =>
		hashmap[a] > hashmap[b] ? a : b
	);
}

export default ApiCall;
