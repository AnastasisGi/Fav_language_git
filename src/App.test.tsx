import React from 'react';
import App from './App';
import mockResponseAxios from '../mocks/mockResponseAxios';
import { render, screen } from '@testing-library/react';
import axios from 'axios';
import ApiCall from './Services/ApiCall';
jest.mock('axios');
const ReactTestRenderer = require('react-test-renderer');


describe('Using the testing library of react ', () => {
	test('renders header component', () => {
		render(<App />);
		let text = screen.getByText('Git Language Investigator');
		expect(text).toBeInTheDocument();
	});
});

describe('Using the snapshot', () => {
	test('it tests that the app renders correctly  ', () => {
    const tree = ReactTestRenderer.create(<App />).toJSON();
		console.log('tree', tree);
		expect(tree).toMatchSnapshot();
	});
});

describe('Using axios mock', () => {
  test('the ApiCall returns the fav language from github call', async () => {
    const mockedAxios = axios as jest.Mocked<typeof axios>;
		mockedAxios.get.mockImplementationOnce(() =>
			Promise.resolve(mockResponseAxios)
		);
		await expect(ApiCall('user')).resolves.toEqual('Ruby');
	});
});
