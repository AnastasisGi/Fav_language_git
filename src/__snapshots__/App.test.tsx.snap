// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the snapshot it tests that the app renders correctly   1`] = `
<div
  className="App"
>
  <header>
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-13aa16l-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h1
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Git Language Investigator
        </h1>
      </div>
    </header>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="username-input"
        id="username-input-label"
      >
        User Name
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="username-input"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              User Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <br />
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Submit
    </button>
  </form>
  <br />
</div>
`;

exports[`Using the snapshot renders the Form component   1`] = `
<div
  className="App"
>
  <header>
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-13aa16l-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h1
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Git Language Investigator
        </h1>
      </div>
    </header>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="username-input"
        id="username-input-label"
      >
        User Name
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="username-input"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              User Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <br />
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Submit
    </button>
  </form>
  <br />
</div>
`;

exports[`renders the Form component   1`] = `
<div
  className="App"
>
  <header>
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-13aa16l-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h1
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Git Language Investigator
        </h1>
      </div>
    </header>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="username-input"
        id="username-input-label"
      >
        User Name
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="username-input"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              User Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <br />
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Submit
    </button>
  </form>
  <br />
</div>
`;
